pkg/tfbridge/provider.go

```diff
@@ -59,6 +59,70 @@ type Resource struct {
 	TFName string           // the Terraform resource name.
 }
 
+// collectImporterAttributes runs the Terraform Importer defined on the Resource for the given
+// resource ID, and adds any non-ID attributes to the inputs map.
+func (res *Resource) collectImporterAttributes(resourceID resource.ID, inputs map[string]string) error {
+	// There is nothing to do here if the resource doesn't have an importer defined in the
+	// Terraform schema.
+	if res.TF.Importer == nil {
+		return nil
+	}
+
+	id := resourceID.String()
+
+	glog.V(9).Infof("%s has TF Importer", res.TFName)
+
+	// Prepare a Terraform ResourceData for the importer
+	data := res.TF.Data(nil)
+	data.SetId(id)
+	data.SetType(res.TFName)
+
+	results, err := res.TF.Importer.State(data, nil)
+	if err != nil {
+		return errors.Wrapf(err, "importing %s", id)
+	}
+
+	if len(results) < 1 {
+		return errors.Errorf("importer for %s returned no resources", id)
+	}
+
+	// Find the resource with the matching ID, since an importer can return several (for example,
+	// a security group and associated rules), and set any non-id attributes as inputs to read.
+	for _, result := range results {
+		if result.Id() != id {
+			continue
```

Comment:
There are several Importers which change the ID.  This seems to be used as a way to allow multiple ID formats from users, while normalizing down to the one format understood by the provider.  This seems a useful thing to support from Pulumi in the same way (and users may expect it to work in the same way).

I think we should at least allow the case where only a single resource is returned to just assume that is the desired return value, and potentially also if there is a single resource of the desired type.

Reply:
For example: https://github.com/terraform-providers/terraform-provider-aws/blob/c5f4c2c84362e81dcb9787aa1e2303891290c2cb/aws/resource_aws_api_gateway_gateway_response.go#L32.  But there are many others.

Reply:
I think this is straightforward enough to support - I've taken a shot in b13b8a1.

Reply:
For now I've erred on the side of caution and taken a single result of the correct type as likely being correct, but required a Type _and_ ID match if there are multiple resources returned. This is something I think we should revisit if it becomes an issue rather than relaxing immediately, though I'm open to alternative views!